core/src/io/anuke/mindustry/ui/dialogs/SandboxDialog.java

```diff
@@ -0,0 +1,48 @@
+package io.anuke.mindustry.ui.dialogs;
+
+import io.anuke.arc.scene.ui.Image;
+import io.anuke.arc.scene.ui.Tooltip;
+import io.anuke.arc.scene.ui.layout.Table;
+import io.anuke.mindustry.graphics.Pal;
+import io.anuke.mindustry.type.ContentType;
+import io.anuke.mindustry.type.Mech;
+
+import static io.anuke.mindustry.Vars.*;
+
+public class SandboxDialog extends FloatingDialog{
+    private Table table;
+
+    public SandboxDialog(){
+        super("$sandbox.options");
+        setFillParent(true);
+        shown(this::setup);
+        onResize(this::setup);
+        addCloseButton();
+    }
+
+    private void setup(){
+        cont.clear();
+        cont.pane(m->table=m);
+        table.top();
+        table.margin(10f);
+        table.addButton("$sandbox.rules", ()->ui.custom.dialog.show()).width(300f).center();
```

Comment:
Whenever you use a lambda expression, add spaces before and after the parameters or `()`. This should be
 `() -> ui.custom.dialog.show()`. The spaces are missing in many different places here.